pages/1.11/security/ent/secrets/secrets-api/index.md

```diff
@@ -94,11 +97,42 @@ curl -H "Authorization: token=$(dcos config show core.dcos_acs_token)"
 Authentication tokens expire after five days by default. If your program needs to run longer than five days, you will need a service account. Please see [Provisioning custom services](/1.11/security/ent/service-auth/custom-service-auth/) for more information.
 
 
+## Text and binary secrets
+
+From DC/OS Enterprise version 1.11 the Secrets API supports text and binary secrets. A text secret is a text value encoded with UTF-8 encoding. The Secrets API transparently passes through this value to a secret store backend. A binary secret is automatically encoded with base64 codec before it is passed to a secrets store backend.
+
+Use cases for secret types:
+
+* **Text**: can support storing text passwords, JSON documents, YAML documents
+
+* **Binary**: All the text secrets and also binary files, like JAVA keystore, Kerberos keytab files
+
+**Note**: It is not recommended to use binary secrets values as environment variables. The POSIX environment variables cannot contain NUL bytes.
```

Comment:
Why is it not recommended? is it because it might break because of the POSIX reason?

Should NUL be in backticks?

Reply:
Its just not possible put a binary data with `NUL` bytes into a environment variable because the `NUL` byte is a terminator for value.